tests/unit_tests/utils/viz_migration/treemap_migration_test.py

```diff
@@ -0,0 +1,93 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import json
+
+from superset.app import SupersetApp
+from superset.utils.migrate_viz import get_migrate_class, MigrateVizEnum
+
+treemap_form_data = """{
+  "adhoc_filters": [
+    {
+      "clause": "WHERE",
+      "comparator": [
+        "Edward"
+      ],
+      "expressionType": "SIMPLE",
+      "filterOptionName": "filter_xhbus6irfa_r10k9nwmwy",
+      "isExtra": false,
+      "isNew": false,
+      "operator": "IN",
+      "operatorId": "IN",
+      "sqlExpression": null,
+      "subject": "name"
+    }
+  ],
+  "color_scheme": "bnbColors",
+  "datasource": "2__table",
+  "extra_form_data": {},
+  "granularity_sqla": "ds",
+  "groupby": [
+    "state",
+    "gender"
+  ],
+  "metrics": [
+    "sum__num"
+  ],
+  "number_format": ",d",
+  "order_desc": true,
+  "row_limit": 10,
+  "time_range": "No filter",
+  "timeseries_limit_metric": "sum__num",
+  "treemap_ratio": 1.618033988749895,
+  "viz_type": "treemap"
+}
+"""
+
+treemap_processor = get_migrate_class[MigrateVizEnum.treemap]
+
+
+def test_treemap_migrate(app_context: SupersetApp) -> None:
```

Comment:
Maybe have separated tests for upgrade and downgrade?

Reply:
the method of downgrade needs an updated slice so combine upgrade and downgrade in the same unit test case.